stdlib/public/Platform/CMakeLists.txt

```diff
@@ -68,11 +69,28 @@ foreach(sdk ${SWIFT_SDKS})
               "-DGLIBC_INCLUDE_PATH=${GLIBC_INCLUDE_PATH}"
               "-DGLIBC_ARCH_INCLUDE_PATH=${GLIBC_ARCH_INCLUDE_PATH}")
 
+      list(APPEND glibc_modulemap_target_list ${glibc_modulemap_target})
+
+      # If cross-compiling, generate a native modulemap without the sysroot prefix
+      # The prefixed one is still the 'glibc.modulemap' which we'll look at during the build process
+      if(CMAKE_CROSSCOMPILING)
```

Comment:
I'm afraid this is not the right variable to key off.  CMake does not always know that we are cross-compiling.

Instead, you should be checking the `sdk` and `arch` variables set above.